### PR TITLE
ENH MPI shims

### PR DESCRIPTION
**Pull request by beckermr**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/mpi-shims/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  name: mpi-shims
+
+build:
+  skip: True  # [win]
+  number: {{ build }}
+
+outputs:
+  - name: mpich
+    version: "{{ mpich_version }}"
+    string: "external_{{ build }}"
```
[the remark this answers is not in the document]
Comment (beckermr) [reply recipes/mpi-shims/meta.yaml]: Thanks! I addd this for both openmpi and mpich.

**Comment by beckermr** [comment]
@minrk @isuruf do you all have any ideas on how to automatically generate the version list in the recipe?

**Comment by beckermr** [comment]
Ohhh sorry. I wasn’t clear. What I mean is generating the versions from the conda package index in the recipe. Right now when we ship a new version of mpich, someone will have to update this recipe by hand. I want that process automated.

**Comment by beckermr** [comment]
Alright. Well unless you all have a better idea on how to make this more maintainable, I vote we merge. 

Care to approve @isuruf @minrk ?

**Comment by minrk** [comment]
> What I mean is generating the versions from the conda package index when the recipe is built

MPI releases are pretty infrequent, so this shouldn't be a huge burden (and host MPIs tend to be *very* slow to upgrade). The alternative I see is to create these from the 'real' provider feedstocks, so that they automatically get new builds when new versions are released.

**Comment by beckermr** [comment]
Right. I actually tried this on the mpich feedstock. Conda build got confused because two outputs has the same name. Unless we expand the build matrix, I don’t think that option is going to work. :(

**Comment by minrk** [comment]
I was thinking it would be another build matrix entry with only one build, rather than an extra output for each build. Not sure what's best.

**Comment by beckermr** [comment]
Right. Let me make a new pr on mpich and we can discuss there.

**Comment by beckermr** [comment]
Closing in favor of packaging this with the feedstocks

**Pull request closed without merging**